src/transformers/models/dpt/modeling_dpt.py

```diff
@@ -0,0 +1,1121 @@
+                )
+
+    def forward(self, hidden_states: List[torch.Tensor]) -> List[torch.Tensor]:
+        """
+        Args:
+            hidden_states (`List[torch.FloatTensor]`, each of shape `(batch_size, sequence_length + 1, hidden_size)`):
+                List of hidden states from the backbone.
+        """
+        out = []
+
+        for i, hidden_state in enumerate(hidden_states):
+            # reshape to (B, C, H, W)
+            hidden_state, cls_token = hidden_state[:, 1:], hidden_state[:, 0]
+            batch_size, sequence_length, num_channels = hidden_state.shape
+            size = int(math.sqrt(sequence_length))
+            hidden_state = hidden_state.reshape(batch_size, size, size, num_channels)
+            hidden_state = hidden_state.permute(0, 3, 1, 2).contiguous()
+
+            feature_shape = hidden_state.shape
+            if self.config.readout_type == "project":
+                # reshape to (B, H*W, C)
+                hidden_state = hidden_state.flatten(2).permute((0, 2, 1))
+                readout = cls_token.unsqueeze(1).expand_as(hidden_state)
+                # concatenate the readout token to the hidden states and project
+                hidden_state = self.readout_projects[i](torch.cat((hidden_state, readout), -1))
+                # reshape back to (B, C, H, W)
+                hidden_state = hidden_state.permute(0, 2, 1).reshape(feature_shape)
+            elif self.config.readout_type == "add":
+                hidden_state = hidden_state.flatten(2) + cls_token.unsqueeze(-1)
+                hidden_state = hidden_state.reshape(feature_shape)
+            hidden_state = self.layers[i](hidden_state)
+            out.append(hidden_state)
+
+        return out
+
+
+class DPTReassembleLayer(nn.Module):
+    def __init__(self, config, channels, factor):
+        super().__init__()
+        # projection
+        self.projection = nn.Conv2d(in_channels=config.hidden_size, out_channels=channels, kernel_size=1)
+
+        # up/down sampling depending on factor
+        if factor > 1:
+            self.resize = nn.ConvTranspose2d(channels, channels, kernel_size=factor, stride=factor, padding=0)
+        elif factor == 1:
+            self.resize = nn.Identity()
+        elif factor < 1:
+            # so should downsample
+            self.resize = nn.Conv2d(channels, channels, kernel_size=3, stride=int(1 / factor), padding=1)
+
+    def forward(self, hidden_state):
+        hidden_state = self.projection(hidden_state)
+        hidden_state = self.resize(hidden_state)
+        return hidden_state
+
+
+class DPTFeatureFusionStage(nn.Module):
+    def __init__(self, config):
+        super().__init__()
+        self.layers = nn.ModuleList()
+        for _ in range(len(config.neck_hidden_sizes)):
+            self.layers.append(DPTFeatureFusionLayer(config))
+        self.layers[0].residual_layer1 = None
```

Comment:
uff that looks a bit hacky. Can't we pass a flag `add_residual_layer1` to `DPTFeatureFusionLayer` with 
`self.layers.append(DPTFeatureFusionLayer(config, add_residual_layer1 = (i > 0)))`

Reply:
Turns out it works too when removing this line.

So I won't include it.